sequence_processing_pipeline/Pipeline.py

```diff
@@ -249,23 +249,29 @@ def _validate_sample_sheet(self, sample_sheet_path):
         raise PipelineError('Sample-sheet has the following errors:\n'
                             '\n'.join([str(x) for x in msgs]))
 
-    def generate_sample_information_files(self):
+    def generate_sample_info_files(self, addl_info=None):
         """
         Generate sample-information files in self.output_path.
+        :param addl_info: A df of (sample-name, project-name) pairs.
         :return: A list of paths to sample-information-files.
         """
         if self.mapping_file is not None:
+            # Generate a list of BLANKs for each project.
             df = self.mapping_file[['sample_name', 'project_name']]
-            samples = list(df.to_records(index=False))
-            samples = [x for x in samples if x[0].startswith('BLANK')]
-            projects = list(set([y for x, y in samples]))
         else:
-            samples = []
-            for sample in self.sample_sheet.samples:
-                if sample['Sample_ID'].startswith('BLANK'):
-                    samples.append((sample['Sample_ID'],
-                                    sample['Sample_Project']))
-            projects = list(set([y for x, y in samples]))
+            # Aggregate all data into a DataFrame
+            data = [[x['Sample_ID'], x['Sample_Project']] for
+                    x in self.sample_sheet.samples]
+            df = pd.DataFrame(data, columns=['sample_name', 'project_name'])
+
+        if addl_info is not None:
+            df = pd.concat([df, addl_info],
+                           ignore_index=True).drop_duplicates()
+            df.reindex()
+
+        df = df[df["sample_name"].str.startswith("BLANK") == True]  # noqa
+        samples = list(df.to_records(index=False))
+        projects = set([y for x, y in samples])
```

Comment:
Wondering if it would be better to do something like:
samples = df.sample_name.to_list()
projects = df. project_name.unique()

Reply:
I'll confirm the samples output remains as desired and make the changes.

Reply:
Confirmed that what I need is a list of (sample_name, project_name) tuples, and the above change would just give me a list of sample_names. I'll keep samples as is and change projects.